Analytics/CDMUtilSolution/CDMUtil/Manifest/ManifestHandler.cs

```diff
@@ -205,224 +183,156 @@ static string getDataLocation(CdmEntityDeclarationDefinition eDef, string localR
 
             return dataLocation;
         }
-        static string cdmToSQLDataType(string dataType, bool dateTimeAsString = false)
+        
+        public static List<ColumnAttribute> getColumnAttributes(CdmEntityDefinition entityDefinition, AppConfigurations c, ILogger logger)
         {
+            CdmCollection<CdmAttributeItem> cdmAttributeItem = entityDefinition.Attributes;
 
-            string sqlDataType;
+            IEnumerable<Artifacts> sourceColumnLength = getSourceColumnProperties(entityDefinition.EntityName, c, logger);
 
-            switch (dataType.ToLower())
+            List<ColumnAttribute> columnAttributes = new List<ColumnAttribute>();
+            
+            foreach (var attributeItem in cdmAttributeItem)
             {
-                case "biginteger":
-                case "int64":
-                case "bigint":
-                    sqlDataType = "bigInt";
-                    break;
-                case "smallinteger":
-                case "int":
-                case "int32":
-                    sqlDataType = "bigInt";
-                    break;
-                case "date":
-                case "datetime":
-                case "datetime2":
-                    if (dateTimeAsString)
-                    {
-                        sqlDataType = "nvarchar(30)";
-                    }
-                    else
+                CdmTypeAttributeDefinition cdmAttribute = (CdmTypeAttributeDefinition) attributeItem;
+                
+                ColumnAttribute columnAttribute = new ColumnAttribute();
+                columnAttribute.name = cdmAttribute.Name;
+                columnAttribute.description = cdmAttribute.Description;
+                columnAttribute.dataType = cdmAttribute.DataType == null ? cdmAttribute.DataFormat.ToString() : cdmAttribute.DataType.NamedReference;
```

Comment:
Can 'cdmAttribute.DataFormat.ToString()' be null as well?
This will throw an error if it is